### PR TITLE
A handful of fixes for src/network.c

### PR DESCRIPTION
**Pull request by OnkelUlla**
These lines caught my eyes while reviewing pull request #522.

### CODE REVIEW

File: src/network.c [hunk 1]
```diff
@@ -75,6 +75,8 @@ static gboolean transfer(RaucTransfer *xfer, GError **error)
 	char errbuf[CURL_ERROR_SIZE];
 	gboolean res = FALSE;
 
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
```
Comment (ejoerns) [inline src/network.c]: If we add an input check for `error`, we should add one for `xfer`, too, to remain consistent.

``` diff
+	g_return_val_if_fail(xfer, FALSE);
```

File: src/network.c [hunk 2]
```diff
@@ -129,6 +131,8 @@ gboolean download_file(const gchar *target, const gchar *url, gsize limit, GErro
 	gboolean res = FALSE;
 	GError *ierror = NULL;
 
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
```
Comment (ejoerns) [inline src/network.c]: Input checks for `target` and `url` should be added, too.

File: src/network.c [hunk 3]
```diff
@@ -195,6 +199,8 @@ gboolean download_mem(GBytes **data, const gchar *url, gsize limit, GError **err
 	GError *ierror = NULL;
 	char *dl_data = NULL;
 
+	g_return_val_if_fail(error == NULL || *error == NULL, FALSE);
```
Comment (ejoerns) [inline src/network.c]: Input checks for data and url should be added, too.

File: src/network.c [hunk 4]
```diff
@@ -104,15 +104,13 @@ static gboolean transfer(RaucTransfer *xfer, GError **error)
 	r = curl_easy_perform(curl);
 	if (r == CURLE_HTTP_RETURNED_ERROR) {
 		g_set_error(error, G_IO_ERROR, G_IO_ERROR_FAILED, "HTTP returned >=400");
-		res = FALSE;
```
Comment (ejoerns) [inline src/network.c]: While checking if we could replace each goto's with a `curl_easy_cleanup(curl)` + direct return, I stumbled across line 82 where we have a `goto out` without setting `error` before. We should fix this, too, either here or in a follow-up PR.

**Comment by OnkelUlla** [comment]
While already in the act of adding the requested sanity checks I also added them for `download_file_checksum()`.

File: src/network.c (hunk 1 above)
Comment (OnkelUlla) [reply src/network.c]: Done.

File: src/network.c (hunk 2 above)
Comment (OnkelUlla) [reply src/network.c]: Done.

File: src/network.c (hunk 3 above)
Comment (OnkelUlla) [reply src/network.c]: Done.

File: src/network.c (hunk 4 above)
Comment (OnkelUlla) [reply src/network.c]: Done.

File: src/network.c (hunk 3 above)
Comment (ejoerns) [reply src/network.c]: The check for `data` should be extended to:
``` c
g_return_val_if_fail(data && *data == NULL, FALSE);
```

Comment (OnkelUlla) [reply src/network.c]: Done - I left that out intentionally as there are no explicit guarantees defined anywhere with respect to `*data`.

**Comment by ejoerns** [comment]
I've rebased onto master to fix trivial conflict with merged PR #503

**Pull request merged**